### PR TITLE
[Debugger] Log telemetry for async ops that stall out

### PR DESCRIPTION
**Pull request by jstedfast**
Needed for https://devdiv.visualstudio.com/DevDiv/_workitems/edit/802365/

### CODE REVIEW

**Comment by jstedfast** [comment]
This is still a work-in-progress. Need to figure out how to get access to the project flavor guids to log those as well. And maybe some other things?

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebuggingService.cs [hunk 1]
```diff
@@ -905,6 +905,23 @@ static async void OnBusyStateChanged (object s, BusyStateEventArgs args)
 			await Runtime.RunInMainThread (delegate {
 				busyEvaluator.UpdateBusyState (args);
 				if (args.IsBusy) {
+					var session = (DebuggerSession) s;
+
+					if (sessions.TryGetValue (session, out var manager)) {
+						var metadata = new Dictionary<string, object> {
+							["DebuggerType"] = manager.Engine.Id,
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebuggingService.cs]: Seems like we could possibly send an event without the manager type if not found.

Comment (jstedfast) [reply main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebuggingService.cs]: From what I can tell, it shouldn't ever fail. I should prob have just done `var manager = sessions[session];`

**Comment by jstedfast** [comment]
Related: https://github.com/xamarin/md-addins/pull/4971

**Comment by sgmunn** [comment]
@jstedfast build failures

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebuggingService.cs [hunk 2]
```diff
@@ -905,6 +905,23 @@ static async void OnBusyStateChanged (object s, BusyStateEventArgs args)
...
+							["Debugger.AsyncOperation.Description"] = args.Description,
+							["Debugger.EvaluationOptions.AllowDisplayStringEvaluation"] = args.EvaluationContext.Options.AllowDisplayStringEvaluation,
+							["Debugger.EvaluationOptions.AllowMethodEvaluation"] = args.EvaluationContext.Options.AllowMethodEvaluation,
+							["Debugger.EvaluationOptions.AllowTargetInvoke"] = args.EvaluationContext.Options.AllowTargetInvoke,
+							["Debugger.EvaluationOptions.AllowToString"] = args.EvaluationContext.Options.AllowToStringCalls,
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebuggingService.cs]:
```suggestion
							["Debugger.EvaluationOptions. AllowToStringCalls"] = args.EvaluationContext.Options.AllowToStringCalls,
```

If we want to mirror the property name. The rest of them do, seemed inconsistent

**Comment by sgmunn** [comment]
tests were green but not reported.

**Pull request merged**